platform/test/spec/testActivityConfigValidatorSpec.js

```diff
@@ -0,0 +1,84 @@
+/*global describe, it, expect, beforeEach --  functions provided by Jasmine */
+import { ActivityConfigValidator } from "../../src/ActivityConfigValidator.js"
+import { ACTIVITY_2PANELS_1ACTION } from "../resources/TestActivityFiles.js";
+import { ConfigValidationError } from "../../src/ConfigValidationError.js"
+
+
+const EXPECTED_FILE_TYPE = "ActivityConfig"; 
+
+describe("ActivityConfigValidator", () => {
+      
+    describe("constructor", () => {
+        it("can be created", () => {
+            // Call the target object
+            let acv = new ActivityConfigValidator();
+            
+            // Check the expected results
+            expect(acv).toBeInstanceOf(ActivityConfigValidator);
+        })
+    })
+
+    describe("validate tool configration", () => {
+        let acv;
+        let activityConfig;
+        
+        //Setup
+        beforeEach( () => {
+            activityConfig = JSON.parse(ACTIVITY_2PANELS_1ACTION);
+            acv = new ActivityConfigValidator();
+        })
+
+        it("reports no errors for a valid config", () => {
+            // Call the target object
+            let errors = acv.validateConfigFile(activityConfig);
+            
+            // Check the expected results
+            expect(errors).toHaveSize(0);
+        })
+
+        it("returns errors that are  ConfigValidationError instances",() => {
+            delete activityConfig.activities[0].id;
+
+            // Call the target object
+            let errors = acv.validateConfigFile(activityConfig);
+            let e = errors[0];
+            
+            // Check the expected results
+            expect(e).toBeInstanceOf(ConfigValidationError);
+            expect(e.fileType).toEqual(EXPECTED_FILE_TYPE);
+        })
+
+        it("returns an error if the config has no id key", () => {
+            delete activityConfig.activities[0].id;
+
+            // Call the target object
+            let errors = acv.validateConfigFile(activityConfig);
+            let e = errors[0];
+            
+            // Check the expected results
+            expect(errors).toHaveSize(1);
+            checkErrorPopulated(e);
+        })
+
+        it("returns an error if a panel in config has no id key", () => {
+            delete activityConfig.activities[0].panels[1].id;
+
+            // Call the target object
+            let errors = acv.validateConfigFile(activityConfig);
+            let e = errors[0];
+            
+            // Check the expected results
+            expect(errors).toHaveSize(1);
+            checkErrorPopulated(e);
+        })
+    }) 
+})
+
+function checkErrorPopulated(error){
```

Comment:
These checks should be more specific and test that the correct errors are reported.

Reply:
Improved the `CheckErrorPopulated()` function to test each attribute of the error and updated tests.  Fixed in cb677e9410c8154ce39faa02c77f690e60f89964.